plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/blobstore/AzureBlobContainer.java

```diff
@@ -70,6 +70,11 @@ public boolean blobExists(String blobName) {
     @Override
     public InputStream readBlob(String blobName) throws IOException {
         logger.trace("readBlob({})", blobName);
+
+        if (!blobExists(blobName)) {
+            throw new IOException("Blob [" + blobName + "] does not exist");
+        }
+
```

Comment:
I forgot, Azure was ambiguous about the behavior, so its good to keep the blobExists check as you have it